trollhunter-unity/Assets/Scripts/TrollBehavior.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class TrollBehavior : MonoBehaviour {
+
+	public static readonly List<long> LevelToExperience = new List<long> {0, 100, 200, 300, 400};
+
+	public enum LifeCycle {
+		EGG,
+		BABY,
+		ADULT,
+	}
+
+	public struct InternalParameter {
+		public long experience;
+	}
+
+	private LifeCycle lifeCycle;
+
+	private InternalParameter parameter;
+
+	private TrollPhenotypes phenotypes;
+
+	public void UpdatePhenotype() {
+		var parts = phenotypes.GetAvailableParts(lifeCycle);
+		// Implement mapping function here.
+		Dictionary<string, string> stub = new Dictionary<string, string> {
+			{"eye", "mean"},
+			{"feet", "tiny"},
+		};
+		phenotypes.RenderParts(stub);
+	}
+
+	public long GetLevel() {
+		return LevelToExperience.IndexOf(LevelToExperience.First(x => x <= parameter.experience));

```

Comment:
Sorry yeah, this line is better as
return LevelToExperience.FindIndex(x => x >= parameter.experience);